dlc/src/main/scala/org/bitcoins/dlc/execution/DLCExecutor.scala

```diff
@@ -85,33 +85,74 @@ case class DLCExecutor(signer: DLCTxSigner)(implicit ec: ExecutionContext) {
     */
   def executeDLC(
       dlcSetup: SetupDLC,
-      oracleSigs: Vector[OracleSignatures]): Future[ExecutedDLCOutcome] = {
+      oracleSigs: Vector[OracleSignatures]): ExecutedDLCOutcome = {
+    val remoteFundingPubKey = if (isInitiator) {
+      builder.acceptFundingKey
+    } else {
+      builder.offerFundingKey
+    }
+
+    DLCExecutor.executeDLC(dlcSetup.cets,
+                           oracleSigs,
+                           signer.fundingKey,
+                           remoteFundingPubKey,
+                           builder.contractInfo,
+                           dlcSetup.fundingTx)
+  }
+
+  def executeRefundDLC(dlcSetup: SetupDLC): RefundDLCOutcome = {
+    val SetupDLC(fundingTx, _, refundTx) = dlcSetup
+    RefundDLCOutcome(fundingTx, refundTx)
+  }
+}
+
+object DLCExecutor {
 
-    val threshold = dlcSetup.cets.head._1.oracles.length
+  def executeDLC(
```

Comment:
Could you add a scaladoc on this method for what the behavior is when we find a signature that does not correspond to an outcome? 

I have a hard time understanding what is happening in these scenarios, as we should be able to always recover the outcome from the signature -- at least that is my understanding?

Specifically i'm talking about https://github.com/nkohen/bitcoin-s-core/blob/ac4aa8edd6063e726e54567a0c72e93a8be74c89/dlc/src/main/scala/org/bitcoins/dlc/execution/DLCExecutor.scala#L132